### PR TITLE
[PAY-577] Add "in progress" page to Buy Audio Modal

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description
image
image
image
image

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

Tested locally against stage

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

Already behind a feature flag

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-buy-in-progress

File: packages/common/src/utils/formatUtil.ts [hunk 1]
```diff
@@ -175,3 +175,27 @@ export const parseWeiNumber = (number: string) => {
     return null
   }
 }
+
+type FormatOptions = {
+  minDecimals?: number
+  maxDecimals?: number
+  excludeCommas?: boolean
+}
```
Comment (dylanjeffers) [inline packages/common/src/utils/formatUtil.ts:183]: spaces plz :)

File: packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  purchasingSol: 'Purchasing SOL',
+  convertingSol: 'Converting SOL to $AUDIO',
+  moreInfo: 'More Info',
+  lessInfo: 'Less Info',
+  usd: 'USD',
+  sol: 'SOL',
+  audio: '$AUDIO'
+}
+
+export const InProgressPage = () => {
```
Comment (dylanjeffers) [inline packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx:31]: are these pages? i think it's fine but wonder if there is a good alternative?

File: packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  const purchaseInfo = useSelector(getAudioPurchaseInfo)
+  const buyAudioFlowStage = useSelector(getBuyAudioFlowStage)
+  const isStepOne = buyAudioFlowStage === BuyAudioStage.PURCHASING
+  let firstToken, secondToken
```
Comment (dylanjeffers) [inline packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx:35]: type these potentially? also i think to do that it should be separate lets, which i also think is more standard

File: packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+          {isStepOne ? messages.step1 : messages.step2}
+        </span>
+      </div>
+      {isStepOne ? (
+        <div className={styles.callToAction}>
+          {messages.completeWithCoinbase}
+        </div>
+      ) : null}
+      <ToggleCollapseButton
+        id='buy-audio-more-info'
```
Comment (dylanjeffers) [inline packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx:83]: so i think the button itself shouldn't have the id, but it should "aria-controls" something with that id?

Comment (dylanjeffers) [reply packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx:83]: haven't looked at implementation too closely, but i might do: 
```
<CollapsibleSection aria-controls="content-id"><div id="content-id" .../> </CollapsibleSection />
```

Prefer that naming cause it's not really a Button here, more like an accordion.
- Other option is to pass id like we are doing and derive the relationship within the CollapsibleSection component, guessing that's what's currently going on.
- Or have lodash create a unique id for it?

**Review by dylanjeffers — APPROVED** [review]
lookin sweet

File: packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx (hunk 4 above)
Comment (rickyrombo) [reply packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx:83]: The button wraps the collapsible contents in a div with that ID, so I think that's effectively option 2 you have listed?

File: packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx (hunk 2 above)
Comment (rickyrombo) [reply packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx:31]: how about `InProgressModalPage`?

File: packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx (hunk 4 above)
Comment (dylanjeffers) [reply packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx:83]: yeah thats what i was guessing.... okay suppose that's fine! but yeah what do we think about `<CollapsibleSection />` or something of that naming?

File: packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx (hunk 2 above)
Comment (dylanjeffers) [reply packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx:31]: right... well it's technically modal-content right? so maybe that?

Comment (dylanjeffers) [reply packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx:31]: but also down for `InProgressContent`?

File: packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx (hunk 4 above)
Comment (rickyrombo) [reply packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx:83]: Yeah I like that - maybe just `<Collapsible>`

File: packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx (hunk 2 above)
Comment (rickyrombo) [reply packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx:31]: sure, can do that!

File: packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx (hunk 4 above)
Comment (dylanjeffers) [reply packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx:83]: yeah a bit abstract, but cool with it

Comment (rickyrombo) [reply packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx:83]: `Section` makes me think `<section>` haha

File: packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx (hunk 2 above)
Comment (rickyrombo) [reply packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx:31]: actually how about InProgressPageContent? they really are pages of the modal in my opinion

Comment (rickyrombo) [reply packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx:31]: I'm going to leave this as Page for now, can revisit!!!

File: packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx (hunk 4 above)
Comment (rickyrombo) [reply packages/web/src/components/buy-audio-modal/components/InProgressPage.tsx:92]: Opted for `CollapsibleContent`!

File: packages/common/src/utils/formatUtil.ts (hunk 1 above)
Comment (piazzatron) [reply packages/common/src/utils/formatUtil.ts:183]: we should figure out how to encode this as an ESLint rule 👀

**Pull request merged**